src/steamship/agents/context.py

```diff
@@ -0,0 +1,64 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Block
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class AgentContext(BaseModel, ABC):
+    # The Context of an Agent comprises:
+    #  * The chat history of the agent with the user,
+    #  * Explicitly stored long-term data for the user:
+    #    * Fetchable by key, if one is provided
+    #    * Searchable
+    # Context is specific to a single user, for now.
+    # This version makes the concept of "Memory" just part of the implementation of context;
+    #   if we don't like this we can split it out
+
+    @abstractmethod
+    @staticmethod
+    def get_or_create_context(user_id: str) -> "AgentContext":
+        """Get or create a new context for a particular end-user"""
+        pass
+
+    # Chat history (short term memory)
+    # This could be stored in a File object
```

Comment:
I also wonder if we want to parse out and save the Agent's scratch pads in file objects for structured logging of ReAct, tool invocation, etc.